papr/papr2kube.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+# Convert a papr YAML file into a Kubernetes Job
+
+import os
+import sys
+import re
+import time
+import yaml
+import traceback
+import argparse
+import subprocess
+
+# XXX: switch to relative imports when we're a proper module
+from papr import PKG_DIR
+import papr.utils.parser as paprparser
+
+GH_NAME_REGEX = re.compile('^[A-Za-z0-9_.-]+$')
+
+def paprsuite2kubejob(gh_org, gh_repo, commitsha, suiteidx, suite):
+    commitsha_short = commitsha[0:10]
+    name = 'papr-{}-{}-{}-{}'.format(gh_org, gh_repo, commitsha_short, suiteidx)
+    metadata = {'name': name}
+    cmd = 'set -xeuo pipefail\n'
+    for test in suite['tests']:
+        cmd += test + '\n'
+    volumes = [
+        {'name': 'builddir', 'emptyDir': {}}
+    ]
+    containers = [{
+        'name': name,
+        'image': suite['container']['image'],
+        'volumeMounts': [
+            { 'name': 'builddir',
+              'mountPath': '/srv',
+            }
+        ],
+        'securityContext': {'runAsUser': 0},
+        'workingDir': '/srv/build',
+        'command': ["/usr/bin/bash", "-c", cmd],
+    }]
+    initContainers = [
+        { 'name': 'init-git',
+          'image': 'registry.centos.org/centos/centos:7',
+          'volumeMounts': [
+              { 'name': 'builddir',
+                'mountPath': '/srv',
+              }
+          ],
+          'securityContext': {'runAsUser': 0},
+          'workingDir': '/srv/',
+          'command': ['/bin/sh', '-c',
+                      '''set -xeuo pipefail; yum -y install git
+                      git clone --depth=100 https://github.com/{gh_org}/{gh_repo} build
```

Comment:
Obviously this is a hack; we should at least optionally support a git mirroring setup (itself exposed as a kube service).